samza-sql/src/main/java/org/apache/samza/sql/planner/QueryPlanner.java

```diff
@@ -109,15 +124,19 @@ private void registerSourceSchemas(SchemaPlus rootSchema) {
     }
   }
 
-  public RelRoot plan(String query) {
+  private Planner getPlanner() {
```

Comment:
The planner is a closable resource I think it would be better to use it within a try block or make sure to close it when done.

Reply:
Sure. Added code to close it now.